src/AutoMapper/MappingEngine.cs

```diff
@@ -410,40 +382,34 @@ protected override Expression VisitMember(MemberExpression node)
         }
 
 
-        object IMappingEngineRunner.Map(ResolutionContext context)
+        public object Map(ResolutionContext context)
         {
             try
             {
+                if (context.TypeMap != null)
+                {
+                    context.TypeMap.Seal();

```

Comment:
Nah DynamicMap thing is really meant for people that can't be bothered with CreateMap